src/App.tsx

```diff
@@ -1,18 +1,43 @@
 import React from 'react';
+import { tabs } from './api/data';
 
 import './App.scss';
+import { Tabs } from './components/Tabs';
 
-const tabs: Tab[] = [
-  { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
-  { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
-  { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
-];
+type State = {
+  selectedTab: Tab,
+};
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected tab is&nbsp;Tab 1</h1>
-    <div>{tabs.length}</div>
-  </div>
-);
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedTab: tabs[0],
+  };
 
-export default App;
+  onTabSelected = (selectedTab: Tab) => {
+    this.setState({
+      selectedTab: tabs.find(
+        tab => tab.id === selectedTab.id,
+      ) || tabs[0],
+    });
+  };
```

Comment:
Can you use selectedTab like the actual tab?

```suggestion
  onTabSelected = (selectedTab: Tab) => {
    this.setState({
      selectedTab: selectedTab || tabs[0],
    });
  };
```